charts/patch/templates/patch.json

```diff
@@ -35,6 +35,7 @@
     "value": "{{$value}}"
   },
   {{- end }}
+  {{- if .Values.injection }}
```

Comment:
Currently, As full `Values` are rendered both in normal injection and `manual` injection, There is no way for the templates to differentiate that, as they depend on checking whether fields like `.Values.proxy`, etc are nil or not. Hence, that logic is moved into the template by adding a new `.Values.injection` which is set when it is `--manual` or through the Injector.